src/app/modules/submission/email-submission/email-submission.service.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  isProcessedCheckboxResponse,
+  isProcessedTableResponse,
+} from '../../../utils/field-validation/field-validation.guards'
+import { ProcessedFieldResponse } from '../submission.types'
+
+import { EmailData, EmailDataForOneField } from './email-submission.types'
+import {
+  getAnswerForCheckbox,
+  getAnswerRowsForTable,
+  getFormattedResponse,
+} from './email-submission.util'
+
+export const createEmailData = (
+  parsedResponses: ProcessedFieldResponse[],
+  hashedFields: Set<string>,
+): EmailData => {
+  const emailData: EmailData = {
+    autoReplyData: [],
+    jsonData: [],
+    formData: [],
+  }
+  parsedResponses.forEach((response) => {
+    let formattedResponses: EmailDataForOneField[]
+    if (isProcessedTableResponse(response)) {
+      formattedResponses = getAnswerRowsForTable(response).map((row) =>
+        getFormattedResponse(row, hashedFields),
+      )
+    } else if (isProcessedCheckboxResponse(response)) {
+      const checkbox = getAnswerForCheckbox(response)
+      formattedResponses = [getFormattedResponse(checkbox, hashedFields)]
+    } else {
+      formattedResponses = [getFormattedResponse(response, hashedFields)]
+    }
```

Comment:
nit: can extract into a separate function? Gets a little hard to read :P

maybe
```ts
const formattedResponses = formatResponse(response)
```

then we can directly map it into email data, looks cleaner!

Reply:
okay it doesn't map directly on to email data because the shapes are different, but I've extracted a helper function